src/org/javarosa/xform/parse/IXFormParserFunctions.java

```diff
@@ -0,0 +1,17 @@
+package org.javarosa.xform.parse;
+
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.IDataReference;
+import org.javarosa.core.model.IFormElement;
+import org.javarosa.core.model.condition.Condition;
+import org.javarosa.core.model.condition.Recalculate;
+import org.javarosa.xpath.parser.XPathSyntaxException;
+
+/** Provides functions from XFormParser to classes split off from it, to avoid a cyclical dependency. */
```

Comment:
I’m trying to follow a practice where module A and module B each depend on the other. I don’t think it’s necessary as far as the compiler’s concerned. It’s a temporary solution.